### DISCUSSION TITLE
Kuka turntable external axis

### DISCUSSION THREAD

**Discussion by alexmkrolak** [visose/Robots#151] [Q&A] [answered]
I'm having trouble programming our external axis which is a KUKA KP1-V500. It moves fine to all the angles you provide it with, but if you exceed +-180 degrees, then the turntable will then exhibit behaviour where it will go to 360deg when you are telling it to move back to 0 degrees. CAn you think of any ways of overcoming this. Perhaps using custom KRL code?

**Comment by visose** [comment]
Does the Grasshopper simulation match what is happening with the real robot and positioner?

**Reply by alexmkrolak** [reply]
In short, no the simulation doesn't match, no.

Sorry, this took me a while to get back to, but I think the answer to this is that the positioner always takes the shortest path to the specified angle during a LIN command. So if the positioner was at E 210, and I wanted it to go back to E 0, I would have to tell it to go to E 90 or maybe E 91 first, and then to E 0. Otherwise, if I gave it a LIN command with E 0 when it was at E 210 it would just go to E 360.

In the simulation, I believe the turntable would go to E 0 if told to from E 210.

I've yet to find a KRL command that can override this shortest path configuration. Or it could be that because the turntable is set to "endless" rotation, then this happens?

**Reply by visose** [reply]
@alexmkrolak can you double check if the configuration in the .xml file matches the configuration in the robot controller? In particular, if the external axis is set to infinite range, the .xmls `minrange` and `maxrange` properties should have the values `-INF` and `INF`.

If it is correct, could you attach the .xml file so I can try to reproduce the bug?

**Comment by visose** [comment, marked as answer]
Closing this as stale with the likely answer from the thread. The XML needs to match the controller configuration for the external axis. In particular, if the KUKA external axis is configured as infinite range in the controller, the library XML should use `minrange="-INF"` and `maxrange="INF"`. If the simulation still differs from the real positioner on a current version, please reopen with the XML file and the relevant controller configuration.
